From 139377aebaf485a4ab2666802fc9b6b1bbf7502f Mon Sep 17 00:00:00 2001
From: Matt Porter <mporter@linaro.org>
Date: Wed, 5 Feb 2014 14:01:36 -0500
Subject: [PATCH 1/5] iio: add support for pulse width capture devices

Add a channel type to support pulse width capture devices.
These devices capture the timing of a PWM signal based on a
configurable trigger

Signed-off-by: Matt Porter <mporter@linaro.org>
Signed-off-by: Stephen Arnold <sarnold@vctlabs.com>
---
 drivers/iio/industrialio-core.c | 1 +
 include/uapi/linux/iio/types.h  | 1 +
 2 files changed, 2 insertions(+)

diff --git a/drivers/iio/industrialio-core.c b/drivers/iio/industrialio-core.c
index d2b889918c3e..9587a10b92dc 100644
--- a/drivers/iio/industrialio-core.c
+++ b/drivers/iio/industrialio-core.c
@@ -81,6 +81,7 @@ static const char * const iio_chan_type_name_spec[] = {
	[IIO_PH] = "ph",
	[IIO_UVINDEX] = "uvindex",
	[IIO_ELECTRICALCONDUCTIVITY] = "electricalconductivity",
+	[IIO_PULSE] = "pulse",
 };

 static const char * const iio_modifier_names[] = {
diff --git a/include/uapi/linux/iio/types.h b/include/uapi/linux/iio/types.h
index 22e5e589a274..92c8f3f83c05 100644
--- a/include/uapi/linux/iio/types.h
+++ b/include/uapi/linux/iio/types.h
@@ -40,6 +40,7 @@ enum iio_chan_type {
	IIO_PH,
	IIO_UVINDEX,
	IIO_ELECTRICALCONDUCTIVITY,
+	IIO_PULSE,
 };

 enum iio_modifier {
--
2.12.2
